packages/@aws-cdk/aws-appsync/test/appsync.test.ts

```diff
@@ -16,6 +16,78 @@ test('should not throw an Error', () => {
     });
   };
 
-  // Then
+  // THEN
   expect(when).not.toThrow();
 });
+
+test('appsync should configure pipeline when pipelineConfig has contents', () => {
+  // GIVEN
+  const stack = new cdk.Stack();
+
+  // WHEN
+  const api = new appsync.GraphQLApi(stack, 'api', {
+    authorizationConfig: {},
+    name: 'api',
+    schemaDefinitionFile: path.join(__dirname, 'schema.graphql'),
+  });
+
+  new appsync.Resolver(stack, 'resolver', {
+    api: api,
+    typeName: 'test',
+    fieldName: 'test2',
+    pipelineConfig: ['test', 'test'],
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::AppSync::Resolver', {
+    Kind: 'PIPELINE',
+    PipelineConfig: { Functions: [ 'test', 'test' ] },
+  });
+});
+
+test('appsync should configure resolver as unit when pipelineConfig is empty', () => {
+  // GIVEN
+  const stack = new cdk.Stack();
+
+  // WHEN
+  const api = new appsync.GraphQLApi(stack, 'api', {
+    authorizationConfig: {},
+    name: 'api',
+    schemaDefinitionFile: path.join(__dirname, 'schema.graphql'),
+  });
+
+  new appsync.Resolver(stack, 'resolver', {
+    api: api,
+    typeName: 'test',
+    fieldName: 'test2',
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::AppSync::Resolver', {
+    Kind: 'UNIT',
+  });
```

Comment:
worth verifying the `pipelineConfig` property as well since kind is driven by it

Reply:
i'm a little confused?

if `pipelineConfig` is not configured, then `Kind` will be `UNIT`.. if `pipelineConfig` is configured then `Kind` will be `PIPELINE`.

Reply:
i meant verify the resource itself rather than a subset of properties but you're right, this doesn't need to happen right now. marking it as resolved.